src/legacy/ui/public/new_platform/set_services.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { pick } from 'lodash';
+
+import { PluginsSetup, PluginsStart } from './new_platform';
+import { LegacyCoreSetup, LegacyCoreStart } from '../../../../core/public';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import * as dataServices from '../../../../plugins/data/public/services';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import * as visualizationsServices from '../../../../plugins/visualizations/public/services';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { createSavedVisLoader } from '../../../../plugins/visualizations/public/saved_visualizations/saved_visualizations';
+
+interface NpSetup {
+  core: LegacyCoreSetup;
+  plugins: PluginsSetup;
+}
+
+interface NpStart {
+  core: LegacyCoreStart;
+  plugins: PluginsStart;
+}
+
+export function setSetupServices(npSetup: NpSetup) {
+  // Services that need to be set in the legacy platform since the legacy data plugin
+  // which previously provided them has been removed.
+  dataServices.setInjectedMetadata(npSetup.core.injectedMetadata);
+  visualizationsServices.setUISettings(npSetup.core.uiSettings);
+  visualizationsServices.setUsageCollector(npSetup.plugins.usageCollection);
+}
```

Comment:
Most of this file is just relocating stuff from `new_platform.ts` to keep everything clean and remove some duplication.